test/expr/unary/async_await.swift

```diff
@@ -144,3 +144,53 @@ func invalidAsyncFunction() async {
 func validAsyncFunction() async throws {
   _ = try await throwingAndAsync()
 }
+
+// Async let checking
+func mightThrow() throws { }
+
+func getIntUnsafely() throws -> Int { 0 }
+func getIntUnsafelyAsync() async throws -> Int { 0 }
+
+extension Error {
+  var number: Int { 0 }
+}
+
+func testAsyncLet() async throws {
+  async let x = await getInt()
+  print(x) // expected-error{{reference to async let 'x' is not marked with 'await'}}
+  print(await x)
```

Comment:
Compared to the proposals and other docs, this is the first I've seen `await` being used on an `async let` initializer. Is the `await` keyword going to be required for initializers that call async functions? Are there two suspension points, at the initializer and at the subsequent read (on line 161)?

I would think the `await` would only be required in a case like this:

```
async let y = getY()
async let x = await calculateX(y)
```

Reply:
I (intentionally) deviated from the proposal document, because as I was writing the tests I felt like skipping the `await` was arbitrary and confusing.

Reply:
oh, why arbitrary to not have it? My feeling was the opposite, I thought that adding the `await` reads like a suspension point, but if it's not a suspension point then the `await` seems arbitrary. And as I mentioned, the two `await`s also struck me as potentially confusing. I'm not arguing a position, just wanted to understand this use, and to share my perspective, for what it's worth.

Reply:
We should take this over to the forums. You're right that the `async let` itself isn't necessarily a suspension point, so in that sense, the `await` might be misleading. On the other hand, it tells you whether you have asynchronous code within the initializer, which... might?... be interesting.